Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -3669,3 +3693,93 @@ extension BrowserViewController: UNUserNotificationCenterDelegate {
         completionHandler()
     }
 }
+
+extension BrowserViewController: PlaylistHelperDelegate {
+    func present(controller: UIViewController) {
+        present(controller, animated: true)
+    }
+    
+    func showPlaylistToast(info: PlaylistInfo, itemState: PlaylistItemAddedState) {
+        guard Preferences.Playlist.showToastForAdd.value,
+              let selectedTab = tabManager.selectedTab,
+              selectedTab.url?.isPlaylistSupportedSiteURL == true else {
+            return
+        }
+        
+        if let toast = self.playlistToast {
+            toast.item = info
+            return
+        }
+        
+        // Item requires the user to choose whether or not to add it to playlists
+        if itemState == .pendingUserAction {
+            let toast = PlaylistToast(item: info, state: .itemPendingUserAction) { [weak self] buttonPressed in
+                guard let self = self, let info = self.playlistToast?.item else { return }
+                if buttonPressed {
+                    // Update playlist with new items..
+                    Playlist.shared.addItem(item: info, cachedData: nil) {
+                        log.debug("Playlist Item Added")
+                        
+                        self.playlistToast = nil
+                        
+                        DispatchQueue.main.async {
+                            self.showPlaylistToast(info: info, itemState: .added)
+                            UIImpactFeedbackGenerator(style: .medium).bzzt()
+                        }
+                    }
+                } else {
+                    self.playlistToast = nil
+                }
+            }
+            
+            self.playlistToast = toast
+            self.show(toast: toast, afterWaiting: .milliseconds(250), duration: .seconds(10))
+        } else if itemState == .existing {
+            // Item already exists in playlist, so ask them if they want to view it there
```

Comment:
It will only play the first item. It doesn't automatically play the item from the page you're viewing. But also, auto-play depends on a toggle in settings so it may not even auto-play either.